spec/datadog/core/diagnostics/environment_logger_spec.rb

```diff
@@ -33,7 +33,11 @@
 
     before do
       allow(Datadog).to receive(:logger).and_return(tracer_logger)
+      allow(tracer_logger).to receive(:debug?).and_return true
+      allow(tracer_logger).to receive(:debug)
       allow(tracer_logger).to receive(:info)
+      allow(tracer_logger).to receive(:warn)
+      allow(tracer_logger).to receive(:error)
     end
```

Comment:
I'm curious -- why was this and the other similar change on `buffer_spec.rb` needed?

Reply:
The fact that we are building the appsec component when booting up triggers some calls to the logger when booting up the libddwaf dependency:

https://github.com/DataDog/dd-trace-rb/blob/cebb06754a85397fc88f45ff5f172f5364d6d98a/lib/datadog/appsec/processor.rb#L176

https://github.com/DataDog/dd-trace-rb/blob/cebb06754a85397fc88f45ff5f172f5364d6d98a/lib/datadog/appsec/processor.rb#L50

https://github.com/DataDog/dd-trace-rb/blob/cebb06754a85397fc88f45ff5f172f5364d6d98a/lib/datadog/appsec/processor.rb#L182

The tests stubs only the `:info` method so the test was failing with unexpected method call

Reply:
We could have a more ergonomic test helper for mocking `Datadog.logger`.

That could be incorporated into a separate PR.

Reply:
Ack :+1: 

To be honest, a lot of these tests boot up half of the library, which I think they shouldn't (rather than just fixing the logger), but I do agree some of those changes are probably better done as a separate :)